src/System Application/App/Email/src/Message/EmailMessageImpl.Codeunit.al

```diff
@@ -599,6 +599,43 @@ codeunit 8905 "Email Message Impl."
         exit(GlobalEmailMessage.Get(MessageId));
     end;
 
+    procedure DeleteOrphanedMessages(StartMessageId: Guid; MessagesToIterate: Integer) NextMessageId: Guid
+    var
+        EmailMessage: Record "Email Message";
+        EmptyGuid: Guid;
+        MessageNo: Integer;
+    begin
+        EmailMessage.SetLoadFields(Id);
+        EmailMessage.SetFilter(Id, '>=%1', StartMessageId);
```

Comment:
Larger or equal than a guid value? Is this intentional?

Reply:
This wouldn't be right. We use `CreateGuid()` for the Id, so it doesn't guarantee that it's going to be in sequence.

Reply:
Yes, you start with zero guid and work your way up. I know we use CreateGuid but this is for cleaning up your ophaned messages. As Darrick says, we use CreateGuid() so they are not in sequence. That's why we need to start from zero guid and work our way through, we can't start from some clever place as they are all over.